refactor: 全局档案管理系统

Pigeon0v0 opened the pull request:
close #370

Pigeon0v0 committed:
feat(wip): profile management ui

Pigeon0v0 committed:
feat(wip): Profile Management

Pigeon0v0 committed:
feat: 档案管理，但是离线 UUID 没做完

Pigeon0v0 committed:
ui: 微调头像间距和字体大小

Pigeon0v0 committed:
refactor: 离线UUID现在遵循行业规范
feat: 允许正版验证不成功时继续启动
fix: 皮肤显示不正确

EncVar commented:
Closes #370

Pigeon0v0 committed:
fix: str

EncVar committed:
feat: duplicate profile judgement

Pigeon0v0 committed:
Merge pull request #6 from PCL-Community/profile

Pigeon0v0 committed:
feat: 正版验证皮肤更改

Pigeon0v0 committed:
feat: 补全原有档案功能
feat: 支持在启动器内修改正版 ID
feat: 支持与 HMCL 双向传输档案列表

Pigeon0v0 committed:
feat: 尝试做了个列表头像

Pigeon0v0 committed:
refactor: 使用 McProfile 类定义档案信息
feat: 档案列表现在能显示头像了
fix: 修复一个可能导致档案列表呈指数级增长的 BUG
chore: clean up

Pigeon0v0 commented:
应该差不多是好了，看看能不能合了

Pigeon0v0 committed:
Merge branch 'dev' into profile

Pigeon0v0 committed:
fix: 头像少了左边一列像素
fix: 不能正常获取档案描述

Pigeon0v0 committed:
fix #435

Pigeon0v0 commented:
等一手 #396 吧

Pigeon0v0 committed:
fix: 没选档案现在不允许启动游戏了

Pigeon0v0 committed:
feat: 创建离线验证档案时允许选择 UUID 生成方式
dev: 部分档案信息使用新的系统获取

Pigeon0v0 committed:
clean up

Pigeon0v0 committed:
feat: 敏感档案信息加密与解密
feat: 给档案导入导出加个计数
fix: 首次启动时若没有档案会意外允许启动游戏

Pigeon0v0 committed:
chore: 清理旧版档案配置文件
fix: 临时修复加解密字符串为空时会意外报错的问题

Pigeon0v0 committed:
fix: MyListItem Logo 路径检查严格一点
fix: 更换头像判空兜底以避免报错

Pigeon0v0 committed:
feat: 允许锁定实例验证方式
dev: 部分逻辑分离到 ModProfile
dev: clean up

Pigeon0v0 committed:
Merge branch 'dev' into profile

Pigeon0v0 committed:
fix: Authlib 拖拽
dev: clean up

Pigeon0v0 committed:
Merge branch 'profile' of https://github.com/PCL-Community/PCL2-CE-Pigeon into profile

Hill23333 reviewed (changes requested):
经过测试，超级炸，但是不知道 Bug 在哪...

EncVar commented:
> 经过测试，超级炸，但是不知道 Bug 在哪...

哪个部分？

Pigeon0v0 commented:
> 经过测试，超级炸，但是不知道 Bug 在哪...

那你说说 Bug 有些啥啊...

Hill23333 commented:
<details>
<summary>日志</summary>
[15:34:18.657] [Start] 程序版本：2.10.7 (Slow Ring, 367，#cb9bad6fd729695cc8ec60cd4a8fecde9acfb192)
[15:34:18.657] [Start] 识别码：647E-7DF6-B760-BD00
[15:34:18.657] [Start] 程序路径：C:\Users\Hill233\Desktop\Plain Craft Launcher 2.exe
[15:34:18.657] [Start] 系统版本：10.0.19044.0, 架构：X64
[15:34:18.657] [Start] 系统编码：gb2312 (936, GBK=True)
[15:34:18.673] [Start] 管理员权限：False
[15:34:18.876] [System] 获取资源：libwebp64
[15:34:18.907] [Start] 第一阶段加载用时：750 ms
[15:34:19.347] [Start] 版本号从 0 升高到 367
[15:34:19.595] [Start] 最高版本号从 0 升高到 367
[15:34:19.642] [Start] 已解除帮助页面的隐藏
[15:34:19.657] [Start] 全新的 PCL，使用新的版本隔离默认值
[15:34:20.002] [Loader] 加载器 Minecraft Folder List 状态改变：Loading
[15:34:20.002] [Start] 第二阶段加载用时：1031 ms
[15:34:20.048] [Minecraft] 已创建 launcher_profiles.json：C:\Users\Hill233\Desktop\.minecraft\
[15:34:20.048] [Loader] 加载器 Minecraft Folder List 状态改变：Finished
[15:34:20.861] [MyPageRight] 获取到 PanScroll 的值为 null
[15:34:21.032] [System] 窗口已置顶，位置：(1254, 626), 900 x 550
[15:34:21.032] [Start] 第三阶段加载用时：500 ms
[15:34:21.048] [Animation] 动画线程开始
[15:34:21.110] [Launch] 没有已储存的 Minecraft 文件夹
[15:34:21.110] [Loader] 加载器 Minecraft Folder List 状态改变：Loading
[15:34:21.110] [Loader] 加载器 Minecraft Folder List 状态改变：Finished
[15:34:21.142] [Setup] 当前选择的 Minecraft 文件夹：C:\Users\Hill233\Desktop\.minecraft\
[15:34:21.142] [Launch] Minecraft 文件夹：C:\Users\Hill233\Desktop\.minecraft\
[15:34:21.142] [Launch] 当前选择的 Minecraft 版本无效：null
[15:34:21.158] [Loader] 加载器 Minecraft Version List 状态改变：Loading
[15:34:21.189] [Setup] 当前选择的 Minecraft 版本：
[15:34:21.189] [Minecraft] 未找到可用 Minecraft 版本
[15:34:21.189] [Loader] 加载器 Minecraft Version List 状态改变：Finished
[15:34:21.204] [Launch] 无可用 Minecraft 版本
[15:34:21.220] [Profile] 刷新档案列表
[15:34:21.220] [Profile] 开始获取档案列表
[15:34:21.361] 读取档案列表失败：文件“C:\Users\Hill233\AppData\Roaming\.pclce\Profiles.json”正由另一进程使用，因此该进程无法访问此文件。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   在 PCL.ModBase.WriteFile(String FilePath, String Text, Boolean Append, Encoding Encoding)
   在 PCL.ModProfile.GetProfile()
   错误类型：System.IO.IOException
[15:34:21.361] [Update] 未获取到远程版本信息，尝试重新获取
[15:34:21.361] [System] 获取更新信息失败：在 UI 线程中运行
[15:34:21.361] [System] 正在获取版本信息
[15:34:21.361] [System] 启动器为 Fast Ring：False
[15:34:21.361] [System] 使用 Pysio 源获取版本信息
[15:34:22.049] [UI] 弹出提示：暂时无法获取更新信息……
[15:34:22.126] [Animation] 两个动画帧间隔 1078 ms
[15:34:22.158] [System] DPI：96，系统版本：Microsoft Windows NT 10.0.19044.0，PCL 位置：C:\Users\Hill233\Desktop\Plain Craft Launcher 2.exe
[15:34:22.158] [Page] 实例化：清空自定义主页 UI，来源为空
[15:34:22.158] [Minecraft] 启动按钮：无 Minecraft 版本，要求下载
[15:34:22.251] [Control] 普通弹窗：社区版本说明
你正在使用来自 PCL-Community 的 PCL 社区版本，遇到问题请不要向官方仓库反馈！
PCL-Community 及其成员与龙腾猫跃无从属关系，且均不会为您的使用做担保。

如果你是意外下载的社区版，建议下载官方版 PCL 使用。

该版本与官方版本的特性区别：
- 联网通知：暂时没有，在做了在做了.jpg
- 主题切换：不会制作，这是需要赞助解锁的纪念性质的功能
- 百宝箱：部分内容更改和缺失，主线分支没有提供相关内容
[15:34:23.626] [Control] 按下按钮：我知道了
[15:34:23.861] [Control] 普通弹窗：环境警告
PCL CE 不再支持你当前使用的系统，部分功能将无法使用。
PCL CE 要求系统版本至少为 Windows 10 20H2 且安装有 .NET Framework 4.8.1。
非常建议升级到最新版本的 Windows 10 或 Windows 11！
[15:34:24.689] [Control] 按下按钮：我知道了
[15:34:24.908] [Control] 普通弹窗：协议授权
在使用 PCL 前，请同意 PCL 的用户协议与免责声明。
[15:34:26.236] [Control] 按下按钮：同意
[15:34:26.236] [Control] 普通弹框返回：1
[15:34:26.251] [Java] 要求 Java 列表缓存更新
[15:34:26.251] [Java] 初始化未找到可用的 Java，将自动触发搜索
[15:34:26.251] [Loader] 加载器 查找 Java 状态改变：Loading
[15:34:26.283] [Java] 开始部分遍历查找：C:\Users\Hill233\AppData\Roaming\.minecraft\
[15:34:26.283] [Java] 开始完全遍历查找：C:\Users\Hill233\Desktop\
[15:34:26.283] [Java] 开始完全遍历查找：C:\Users\Hill233\Desktop\.minecraft\
[15:34:26.439] [Control] 普通弹窗：出现错误
读取档案列表失败：文件“C:\Users\Hill233\AppData\Roaming\.pclce\Profiles.json”正由另一进程使用，因此该进程无法访问此文件。
   在 PCL.ModBase.WriteFile(String FilePath, String Text, Boolean Append, Encoding Encoding)
   在 PCL.ModProfile.GetProfile()
   错误类型：System.IO.IOException

是否反馈此问题？如果不反馈，这个问题可能永远无法得到解决！
[15:34:26.439] [Java] 开始确认列表 Java 状态，共 0 项
[15:34:26.455] [Loader] 加载器 DlClientList Mojang 状态改变：Loading
[15:34:26.470] [Loader] 加载器 PCL 服务 状态改变：Loading
[15:34:26.470] [System] 正在获取版本信息
[15:34:26.470] [System] 启动器为 Fast Ring：False
[15:34:26.470] [System] 使用 Pysio 源获取版本信息
[15:34:26.470] [Net] 发起网络请求（GET，https://s3.pysio.online/pcl2-ce/announceVer.ini），最大超时 15000
[15:34:26.470] [System] 开始清理任务缓存文件夹
[15:34:26.470] [Net] 获取网络结果：https://launchermeta.mojang.com/mc/game/version_manifest.json，超时 10000ms，要求 Json
[15:34:26.486] [System] 已清理任务缓存文件夹
[15:34:26.486] [Java] Java 搜索完成，发现 0 个 Java
[15:34:26.486] [Loader] 加载器 查找 Java 状态改变：Finished
[15:34:26.532] [Net] 当前代理状态：禁用
[15:34:27.095] [System] 未通过 SSL 证书验证（RemoteCertificateChainErrors），提供的证书为 CN=pysio.online，URL：https://s3.pysio.online/pcl2-ce/announceVer.ini
[15:34:27.658] [Net] 发起网络请求（GET，https://s3.pysio.online/pcl2-ce/announceVer.ini），最大超时 25000
[15:34:27.658] [Net] 当前代理状态：禁用
[15:34:28.001] [System] 未通过 SSL 证书验证（RemoteCertificateChainErrors），提供的证书为 CN=pysio.online，URL：https://s3.pysio.online/pcl2-ce/announceVer.ini
[15:34:28.033] 加载线程 PCL 服务 (19) 出错，已完成 2%：网络请求失败（TrustFailure，基础连接已经关闭: 未能为 SSL/TLS 安全通道建立信任关系。，https://s3.pysio.online/pcl2-ce/announceVer.ini）
   在 PCL.ModNet.NetRequestRetry(String Url, String Method, Object Data, String ContentType, Boolean DontRetryOnRefused, Dictionary`2 Headers)
   在 PCL.ModSecret.UpdateLatestVersionInfo()
   在 PCL.ModSecret.LoadOnlineInfo()
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Net.WebException
→ 基础连接已经关闭: 未能为 SSL/TLS 安全通道建立信任关系。
   在 System.Net.HttpWebRequest.GetResponse()
   在 PCL.ModNet.NetRequestOnce(String Url, String Method, Object Data, String ContentType, Int32 Timeout, Dictionary`2 Headers, Boolean MakeLog, Boolean UseBrowserUserAgent)
   错误类型：System.Net.WebException
→ 根据验证过程，远程证书无效。
   在 System.Net.Security.SslState.StartSendAuthResetSignal(ProtocolToken message, AsyncProtocolRequest asyncRequest, Exception exception)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
   在 System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest, Boolean renegotiation)
   在 System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
   在 System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   在 System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   在 System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   在 System.Net.TlsStream.ProcessAuthentication(LazyAsyncResult result)
   在 System.Net.TlsStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   在 System.Net.PooledStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   在 System.Net.ConnectStream.WriteHeaders(Boolean async)
   错误类型：System.Security.Authentication.AuthenticationException
[15:34:28.033] [Loader] 加载器 PCL 服务 状态改变：Failed
[15:34:28.907] [Net] 获取网络结果：https://zkitefly.github.io/unlisted-versions-of-minecraft/version_manifest.json，超时 10000ms，要求 Json
[15:34:30.017] [Control] 按下按钮：取消
[15:34:30.033] [Control] 普通弹框返回：2
[15:34:30.033] [Profile] 档案列表刷新完成
[15:34:32.175] [Control] 按下图标按钮：BtnPort
[15:34:32.205] [Control] 普通弹窗：导入 / 导出档案
PCL CE 支持导入 HMCL 的全局账户列表，抑或是导出档案列表至 HMCL 全局账户列表。
你想要...？
[15:34:33.454] [Control] 按下按钮：取消
[15:34:33.454] [Control] 普通弹框返回：3
[15:34:33.454] [Profile] 刷新档案列表
[15:34:33.454] [Profile] 开始获取档案列表
[15:34:33.454] 读取档案列表失败：Error reading JObject from JsonReader. Path '', line 0, position 0.
   在 Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   在 Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   在 PCL.ModProfile.GetProfile()
   错误类型：Newtonsoft.Json.JsonReaderException
[15:34:33.454] [Update] 未获取到远程版本信息，尝试重新获取
[15:34:33.454] [System] 获取更新信息失败：在 UI 线程中运行
[15:34:33.454] [System] 正在获取版本信息
[15:34:33.454] [System] 启动器为 Fast Ring：False
[15:34:33.454] [System] 使用 Pysio 源获取版本信息
[15:34:33.970] [UI] 弹出提示：暂时无法获取更新信息……
[15:34:33.970] [Animation] 两个动画帧间隔 500 ms
[15:34:34.032] [Control] 普通弹窗：出现错误
读取档案列表失败：Error reading JObject from JsonReader. Path '', line 0, position 0.
   在 PCL.ModProfile.GetProfile()
   错误类型：Newtonsoft.Json.JsonReaderException

是否反馈此问题？如果不反馈，这个问题可能永远无法得到解决！
[15:34:36.548] [Control] 按下按钮：取消
[15:34:36.548] [Control] 普通弹框返回：2
[15:34:36.548] [Profile] 档案列表刷新完成
[15:34:38.295] [Control] 按下图标按钮：BtnNew
[15:34:38.344] [Control] 选择弹窗：新建档案 - 选择验证类型
[15:34:39.282] [Control] 按下单选框：离线验证
[15:34:39.439] [Net] 获取网络结果：https://zkitefly.github.io/unlisted-versions-of-minecraft/version_manifest.json，超时 30000ms，要求 Json
[15:34:39.845] [Control] 按下按钮：继续
[15:34:39.861] [Control] 选择弹框返回：0
[15:34:40.111] [Control] 输入弹窗：新建档案 - 输入档案名称
[15:34:41.161] [System] 未通过 SSL 证书验证（RemoteCertificateChainErrors），提供的证书为 CN=*.github.io，URL：https://zkitefly.github.io/unlisted-versions-of-minecraft/version_manifest.json
[15:34:41.663] [Net] 获取网络结果：https://zkitefly.github.io/unlisted-versions-of-minecraft/version_manifest.json，超时 4000ms，要求 Json
[15:34:41.944] [System] 未通过 SSL 证书验证（RemoteCertificateChainErrors），提供的证书为 CN=*.github.io，URL：https://zkitefly.github.io/unlisted-versions-of-minecraft/version_manifest.json
[15:34:41.945] [Download] 未列出的版本官方源下载失败: 获取结果失败（9，基础连接已经关闭: 未能为 SSL/TLS 安全通道建立信任关系。，https://zkitefly.github.io/unlisted-versions-of-minecraft/version_manifest.json）
[15:34:41.959] [Loader] 加载器 DlClientList Mojang 状态改变：Finished
[15:34:43.892] [Control] 按下按钮：继续
[15:34:43.907] [Control] 输入弹框返回：114514
[15:34:44.064] [Control] 选择弹窗：新建档案 - 选择 UUID 类型
[15:34:45.173] [Control] 按下单选框：行业规范 UUID（推荐）
[15:34:45.876] [Control] 按下按钮：继续
[15:34:45.876] [Control] 选择弹框返回：0
[15:34:45.892] [Profile] 写入档案列表
[15:34:45.892] [Profile] 开始写入档案，共 1 个
[15:34:45.892] [Profile] 刷新档案列表
[15:34:45.892] [Profile] 开始获取档案列表
[15:34:45.923] [Profile] 获取档案完成，获取到 1 个档案
[15:34:45.923] [Profile] 档案列表刷新完成
[15:34:45.923] [UI] 弹出提示：档案新建成功！
[15:34:47.404] [Control] 按下图标按钮：BtnPort
[15:34:47.412] [Control] 普通弹窗：导入 / 导出档案
PCL CE 支持导入 HMCL 的全局账户列表，抑或是导出档案列表至 HMCL 全局账户列表。
你想要...？
[15:34:48.314] [Control] 按下按钮：取消
[15:34:48.314] [Control] 普通弹框返回：3
[15:34:48.314] [Profile] 刷新档案列表
[15:34:48.314] [Profile] 开始获取档案列表
[15:34:48.314] [Profile] 获取档案完成，获取到 1 个档案
[15:34:48.314] [Profile] 档案列表刷新完成
[15:34:50.094] [Control] 按下图标按钮：BtnNew
[15:34:50.118] [Control] 选择弹窗：新建档案 - 选择验证类型
[15:34:51.017] [Control] 按下单选框：正版验证
[15:34:51.486] [Control] 按下按钮：继续
[15:34:51.486] [Control] 选择弹框返回：1
[15:34:51.501] [Profile] 刷新档案列表
[15:34:51.501] [Profile] 开始获取档案列表
[15:34:51.501] [Profile] 获取档案完成，获取到 1 个档案
[15:34:51.501] [Profile] 档案列表刷新完成
[15:34:52.189] [Control] 按下按钮：开始正版验证
[15:34:52.220] 加载输入获取失败（Loader Login Ms）：无法将类型为“McLoginLegacy”的对象强制转换为类型“McLoginMs”。
   在 Microsoft.VisualBasic.CompilerServices.Conversions.ToGenericParameter[T](Object Value)
   在 PCL.ModLoader.LoaderTask`2.ShouldStart(Object& Input, Boolean IsForceRestart, Boolean IgnoreReloadTimeout)
   错误类型：System.InvalidCastException
[15:34:52.236] [Loader] 加载器 Loader Login Ms 状态改变：Failed
[15:34:52.236] 正版登录尝试失败：无法将类型为“McLoginLegacy”的对象强制转换为类型“McLoginMs”。
   在 Microsoft.VisualBasic.CompilerServices.Conversions.ToGenericParameter[T](Object Value)
   在 PCL.ModLoader.LoaderTask`2.Start(Object Input, Boolean IsForceRestart)
   在 PCL.PageLoginMs._Lambda$__2-0()
   错误类型：System.InvalidCastException
[15:34:52.252] [Control] 普通弹窗：出现错误
正版登录尝试失败：无法将类型为“McLoginLegacy”的对象强制转换为类型“McLoginMs”。
   在 PCL.ModLoader.LoaderTask`2.Start(Object Input, Boolean IsForceRestart)
   在 PCL.PageLoginMs._Lambda$__2-0()
   错误类型：System.InvalidCastException
[15:34:53.736] [Control] 按下按钮：确定
[15:34:54.564] [Control] 按下文本按钮：«  返回
[15:34:54.611] [Profile] 刷新档案界面
[15:34:54.611] [Profile] 使用离线皮肤加载器
[15:34:54.611] [Loader] 加载器 Loader Skin Legacy 状态改变：Loading
[15:34:54.626] [Loader] 加载器 Loader Skin Legacy 状态改变：Finished
[15:34:54.658] 载入头像失败（pack://application:,,,/Plain Craft Launcher 2;component/Images/Skins/Alex.png,Loader Skin Legacy）：未能找到路径“C:\Users\Hill233\AppData\Local\Temp\PCL\Cache\Skin\Head\Legacy_114514_6c03ec9536b9385cb35828d2fe101bc8.png”的一部分。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   在 PCL.MySkin.Load()
   错误类型：System.IO.DirectoryNotFoundException
[15:34:58.705] [Control] 按下图标按钮：BtnSkin
[15:35:00.564] [UI] 弹出提示：当前档案不支持修改披风！
[15:35:02.860] [Control] 按下图标按钮：BtnSelect
[15:35:02.862] [Profile] 刷新档案列表
[15:35:02.862] [Profile] 开始获取档案列表
[15:35:02.862] [Profile] 获取档案完成，获取到 1 个档案
[15:35:02.863] [Profile] 档案列表刷新完成
[15:35:03.764] [Control] 按下图标按钮：BtnNew
[15:35:03.815] [Control] 选择弹窗：新建档案 - 选择验证类型
[15:35:05.173] [Control] 按下单选框：第三方验证
[15:35:05.673] [Control] 按下按钮：继续
[15:35:05.673] [Control] 选择弹框返回：2
[15:35:05.705] [Profile] 刷新档案列表
[15:35:05.705] [Profile] 开始获取档案列表
[15:35:05.705] [Profile] 获取档案完成，获取到 1 个档案
[15:35:05.705] [Profile] 档案列表刷新完成
[15:35:05.939] [System] 诊断信息：
操作系统：Microsoft Windows 10 IoT 企业版 LTSC（32 位：False）
剩余内存：6120 M / 8191 M
DPI：96（100%）
MC 文件夹：C:\Users\Hill233\Desktop\.minecraft\
文件位置：C:\Users\Hill233\Desktop\
[15:35:05.939] 程序出现未知错误：未找到设置项：CacheAuthServerRegister
   在 PCL.ModSetup.Get(String Key, McVersion Version)
   在 PCL.PageLoginAuth.ReloadRegisterButton()
   在 System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   在 System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   在 System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)
   在 System.Windows.BroadcastEventHelper.BroadcastLoadedEvent(Object root)
   在 MS.Internal.LoadedOrUnloadedOperation.DoWork()
   在 System.Windows.Media.MediaContext.FireLoadedPendingCallbacks()
   在 System.Windows.Media.MediaContext.FireInvokeOnRenderCallbacks()
   在 System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   在 System.Windows.Media.MediaContext.RenderMessageHandler(Object resizedCompositionTarget)
   在 System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   在 System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   错误类型：System.Collections.Generic.KeyNotFoundException
[15:35:05.939] [Update] 未获取到远程版本信息，尝试重新获取
[15:35:05.939] [System] 获取更新信息失败：在 UI 线程中运行
[15:35:05.939] [System] 正在获取版本信息
[15:35:05.939] [System] 启动器为 Fast Ring：False
[15:35:05.939] [System] 使用 Pysio 源获取版本信息
[15:35:07.008] [UI] 弹出提示：暂时无法获取更新信息……
[15:35:07.017] [Animation] 两个动画帧间隔 1187 ms
[15:35:08.847] [System] 程序已退出，返回值：Fail

</details>

xinyihl commented:
添加几点：
1.导入/导出档案提示在UI线程执行网络请求
<details>
<summary>详情</summary>

``` 
程序出现未知错误：在 UI 线程执行了网络请求
   在 PCL.ModNet.NetGetCodeByRequestRetry(String Url, Encoding Encode, String Accept, Boolean IsJson, String BackupUrl, Boolean UseBrowserUserAgent)
   在 PCL.ModProfile.MigrateProfile()
   在 PCL.PageLoginProfile.BtnPort_Click()
   在 PCL.MyIconButton.Button_MouseUp(Object sender, MouseButtonEventArgs e)
``` 

</details>

2.离线载入头像失败
3.正版登录尝试失败：索引超出范围
4.第三方登录尝试失败

<details>
<summary>日志</summary>

``` 
[20:56:54.826] [Start] 程序版本：2.10.7 (Slow Ring, 367，#cb9bad6fd729695cc8ec60cd4a8fecde9acfb192)
[20:56:54.837] [Start] 识别码：F89D-6F75-A636-8FDD
[20:56:54.837] [Start] 程序路径：D:\Minecrafts\test.exe
[20:56:54.839] [Start] 系统版本：10.0.22631.0, 架构：X64
[20:56:54.839] [Start] 系统编码：gb2312 (936, GBK=True)
[20:56:54.844] [Start] 管理员权限：False
[20:56:54.872] [System] 获取资源：libwebp64
[20:56:54.895] [Start] 第一阶段加载用时：235 ms
[20:56:55.568] [Loader] 加载器 Minecraft Folder List 状态改变：Loading
[20:56:55.569] [Start] 第二阶段加载用时：609 ms
[20:56:55.632] [Loader] 加载器 Minecraft Folder List 状态改变：Finished
[20:56:56.442] [MyPageRight] 获取到 PanScroll 的值为 null
[20:56:56.571] [System] 窗口已置顶，位置：(423.2, 270.4), 1202.4 x 738.4
[20:56:56.574] [Animation] 动画线程开始
[20:56:56.574] [Start] 第三阶段加载用时：187 ms
[20:56:56.592] [Page] 主页预设：Minecraft 新闻
[20:56:56.592] [Page] 主页自定义数据来源：联网缓存文件
[20:56:56.595] [Loader] 加载器 自定义主页获取 状态改变：Loading
[20:56:56.597] [Launch] Minecraft 文件夹：D:\Minecrafts\.minecraft\
[20:56:56.599] [Net] 获取网络结果：http://pcl.mcnews.thestack.top/version，超时 10000ms
[20:56:56.599] [Java] 缓存中有 1 个可用的 Java：
[20:56:56.601] [Java]  - JDK 21 (21.0.2)：C:\Program Files\Java\jdk-21\bin\
[20:56:56.654] [Profile] 刷新档案列表
[20:56:56.658] [Profile] 开始获取档案列表
[20:56:56.658] [Profile] 获取档案完成，获取到 0 个档案
[20:56:56.658] [Profile] 档案列表刷新完成
[20:56:56.800] [Animation] 两个动画帧间隔 219 ms
[20:56:56.806] [Page] 实例化：加载自定义主页 UI 开始，最终内容长度：54296
[20:56:56.812] [Loader] 加载器 DlClientList Mojang 状态改变：Loading
[20:56:56.815] [Net] 获取网络结果：https://launchermeta.mojang.com/mc/game/version_manifest.json，超时 10000ms，要求 Json
[20:56:56.826] [Loader] 加载器 PCL 服务 状态改变：Loading
[20:56:56.827] [System] 开始清理任务缓存文件夹
[20:56:56.828] [System] 正在获取版本信息
[20:56:56.828] [System] 启动器为 Fast Ring：False
[20:56:56.828] [System] 使用 Pysio 源获取版本信息
[20:56:56.828] [System] 已清理任务缓存文件夹
[20:56:56.830] [Net] 发起网络请求（GET，https://s3.pysio.online/pcl2-ce/announceVer.ini），最大超时 15000
[20:56:56.831] [Net] 当前代理状态：禁用
[20:56:57.221] [Page] 实例化：加载自定义主页 UI 完成
[20:56:57.222] [Animation] 两个动画帧间隔 407 ms
[20:56:57.223] [Minecraft] 启动按钮：Minecraft 版本：D:\Minecrafts\.minecraft\versions\All the Mods 9 - ATM9 OLD\
[20:56:57.524] [Animation] 两个动画帧间隔 282 ms
[20:56:57.560] [System] DPI：120，系统版本：Microsoft Windows NT 10.0.22631.0，PCL 位置：D:\Minecrafts\test.exe
[20:56:57.647] [Loader] 加载器 DlClientList Mojang 状态改变：Finished
[20:56:57.889] [Loader] 加载器 PCL 服务 状态改变：Finished
[20:56:59.771] [Control] 按下图标按钮：BtnNew
[20:56:59.825] [Control] 选择弹窗：新建档案 - 选择验证类型
[20:57:00.898] [Control] 按下按钮：取消
[20:57:00.899] [Control] 选择弹框返回：null
[20:57:00.900] [Profile] 刷新档案列表
[20:57:00.900] [Profile] 开始获取档案列表
[20:57:00.900] [Profile] 获取档案完成，获取到 0 个档案
[20:57:00.900] [Profile] 档案列表刷新完成
[20:57:01.459] [Page] 当前缓存的自定义主页已为最新，当前版本：ea61b02bcbea8cec4435757ba5d7fd6afa4a3a76
，检查源：http://pcl.mcnews.thestack.top/version
[20:57:01.460] [Page] 主页预设：Minecraft 新闻
[20:57:01.460] [Loader] 加载器 自定义主页获取 状态改变：Finished
[20:57:01.460] [Page] 主页自定义数据来源：联网缓存文件
[20:57:01.562] [Control] 按下图标按钮：BtnPort
[20:57:01.577] [Control] 普通弹窗：导入 / 导出档案
PCL CE 支持导入 HMCL 的全局账户列表，抑或是导出档案列表至 HMCL 全局账户列表。
你想要...？
[20:57:02.402] [Control] 按下按钮：导出
[20:57:02.403] [Control] 普通弹框返回：2
[20:57:02.406] [Profile] 刷新档案列表
[20:57:02.406] [Profile] 开始获取档案列表
[20:57:02.407] [Profile] 获取档案完成，获取到 0 个档案
[20:57:02.407] [Profile] 档案列表刷新完成
[20:57:02.415] [UI] 弹出提示：正在导出，请稍后...
[20:57:02.415] [UI] 弹出提示：已导出 0 个档案，部分档案可能需要重新验证密码！
[20:57:04.970] [Control] 按下图标按钮：BtnNew
[20:57:04.993] [Control] 选择弹窗：新建档案 - 选择验证类型
[20:57:06.330] [Control] 按下单选框：离线验证
[20:57:07.001] [Control] 按下按钮：继续
[20:57:07.002] [Control] 选择弹框返回：0
[20:57:07.238] [Control] 输入弹窗：新建档案 - 输入档案名称
[20:57:14.475] [Control] 按下按钮：继续
[20:57:14.477] [Control] 输入弹框返回：xin_yi_hl
[20:57:14.650] [Control] 选择弹窗：新建档案 - 选择 UUID 类型
[20:57:19.676] [Control] 按下单选框：行业规范 UUID（推荐）
[20:57:22.090] [Control] 按下按钮：继续
[20:57:22.090] [Control] 选择弹框返回：0
[20:57:22.093] [Profile] 写入档案列表
[20:57:22.093] [Profile] 开始写入档案，共 1 个
[20:57:22.094] [Profile] 刷新档案列表
[20:57:22.094] [Profile] 开始获取档案列表
[20:57:22.094] [Profile] 获取档案完成，获取到 1 个档案
[20:57:22.096] [Profile] 档案列表刷新完成
[20:57:22.126] [UI] 弹出提示：档案新建成功！
[20:57:25.426] [Control] 按下图标按钮
[20:57:25.451] [Control] 选择弹窗：新建档案 - 选择 UUID 类型
[20:57:26.937] [Control] 按下单选框：行业规范 UUID（推荐）
[20:57:27.730] [Control] 按下按钮：取消
[20:57:27.730] [Control] 选择弹框返回：null
[20:57:29.073] [Control] 按下图标按钮
[20:57:29.104] [Control] 选择弹窗：新建档案 - 选择 UUID 类型
[20:57:29.866] [Control] 按下单选框：行业规范 UUID（推荐）
[20:57:30.458] [Control] 按下按钮：继续
[20:57:30.459] [Control] 选择弹框返回：0
[20:57:30.459] [Profile] 写入档案列表
[20:57:30.459] [Profile] 开始写入档案，共 1 个
[20:57:30.491] [UI] 弹出提示：档案信息已保存！
[20:57:31.426] [Control] 按下图标按钮
[20:57:31.483] [Control] 选择弹窗：新建档案 - 选择 UUID 类型
[20:57:32.098] [Control] 按下单选框：自定义
[20:57:33.002] [Control] 按下按钮：继续
[20:57:33.002] [Control] 选择弹框返回：2
[20:57:33.218] [Control] 输入弹窗：更改档案 xin_yi_hl 的 UUID
[20:57:35.362] [Control] 按下按钮：继续
[20:57:35.363] [Control] 输入弹框返回：27df4c20edce32c18c7f678020360852
[20:57:35.363] [Profile] 写入档案列表
[20:57:35.363] [Profile] 开始写入档案，共 1 个
[20:57:35.383] [UI] 弹出提示：档案信息已保存！
[20:57:39.601] [Control] 按下图标按钮：BtnNew
[20:57:39.607] [Control] 选择弹窗：新建档案 - 选择验证类型
[20:57:40.818] [Control] 按下单选框：正版验证
[20:57:41.690] [Control] 按下单选框：第三方验证
[20:57:42.076] [Control] 按下按钮：继续
[20:57:42.076] [Control] 选择弹框返回：2
[20:57:42.087] [Profile] 刷新档案列表
[20:57:42.088] [Profile] 开始获取档案列表
[20:57:42.088] [Profile] 获取档案完成，获取到 1 个档案
[20:57:42.088] [Profile] 档案列表刷新完成
[20:57:45.498] [Control] 按下按钮：返回
[20:57:45.536] [Profile] 刷新档案界面
[20:57:45.537] [Profile] 使用离线皮肤加载器
[20:57:45.539] [Loader] 加载器 Loader Skin Legacy 状态改变：Loading
[20:57:45.543] [Loader] 加载器 Loader Skin Legacy 状态改变：Finished
[20:57:45.589] 载入头像失败（pack://application:,,,/Plain Craft Launcher 2;component/Images/Skins/Steve.png,Loader Skin Legacy）：未能找到路径“C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\Head\Legacy_xin_yi_hl_27df4c20edce32c18c7f678020360852.png”的一部分。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   在 PCL.MySkin.Load()
   错误类型：System.IO.DirectoryNotFoundException
[20:57:52.562] [Control] 按下图标按钮：BtnSelect
[20:57:52.564] [Profile] 刷新档案列表
[20:57:52.564] [Profile] 开始获取档案列表
[20:57:52.564] [Profile] 获取档案完成，获取到 1 个档案
[20:57:52.565] [Profile] 档案列表刷新完成
[20:57:54.258] [Profile] 选定档案: xin_yi_hl, 以 Legacy 方式验证
[20:57:54.259] [Profile] 刷新档案界面
[20:57:54.259] [Profile] 使用离线皮肤加载器
[20:57:54.259] [Loader] 加载器 Loader Skin Legacy 状态改变：Loading
[20:57:54.259] [Loader] 加载器 Loader Skin Legacy 状态改变：Finished
[20:57:54.282] [Control] 按下单击列表项：xin_yi_hl
[20:57:54.285] 载入头像失败（pack://application:,,,/Plain Craft Launcher 2;component/Images/Skins/Steve.png,Loader Skin Legacy）：未能找到路径“C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\Head\Legacy_xin_yi_hl_27df4c20edce32c18c7f678020360852.png”的一部分。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   在 PCL.MySkin.Load()
   错误类型：System.IO.DirectoryNotFoundException
[20:59:52.778] [Control] 按下图标按钮：BtnSkin
[20:59:55.741] [UI] 弹出提示：当前档案不支持修改披风！
[20:59:57.130] [Control] 按下图标按钮：BtnSkin
[20:59:59.241] [UI] 选择文件返回：
[21:00:00.194] [Control] 按下图标按钮：BtnSkin
[21:00:01.233] [UI] 弹出提示：当前档案不支持修改皮肤！
[21:00:02.577] [Control] 按下图标按钮：BtnSkin
[21:00:07.452] [UI] 选择文件返回：C:\Users\xinyihl\Desktop\Steve.png
[21:00:07.454] [UI] 弹出提示：皮肤保存成功！
[21:00:11.490] [Control] 按下图标按钮：BtnSelect
[21:00:11.492] [Profile] 刷新档案列表
[21:00:11.492] [Profile] 开始获取档案列表
[21:00:11.492] [Profile] 获取档案完成，获取到 1 个档案
[21:00:11.493] [Profile] 档案列表刷新完成
[21:00:12.697] [Profile] 选定档案: xin_yi_hl, 以 Legacy 方式验证
[21:00:12.698] [Profile] 刷新档案界面
[21:00:12.698] [Profile] 使用离线皮肤加载器
[21:00:12.698] [Loader] 加载器 Loader Skin Legacy 状态改变：Loading
[21:00:12.698] [Loader] 加载器 Loader Skin Legacy 状态改变：Finished
[21:00:12.722] [Control] 按下单击列表项：xin_yi_hl
[21:00:12.725] 载入头像失败（pack://application:,,,/Plain Craft Launcher 2;component/Images/Skins/Steve.png,Loader Skin Legacy）：未能找到路径“C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\Head\Legacy_xin_yi_hl_27df4c20edce32c18c7f678020360852.png”的一部分。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   在 PCL.MySkin.Load()
   错误类型：System.IO.DirectoryNotFoundException
[21:00:13.897] [Control] 按下图标按钮：BtnSelect
[21:00:13.898] [Profile] 刷新档案列表
[21:00:13.898] [Profile] 开始获取档案列表
[21:00:13.898] [Profile] 获取档案完成，获取到 1 个档案
[21:00:13.899] [Profile] 档案列表刷新完成
[21:00:16.434] [Control] 按下图标按钮
[21:00:16.440] [Control] 普通弹窗：删除档案确认
你正在选择删除此档案，该操作无法撤销。
确定继续？
[21:00:17.331] [Control] 按下按钮：继续
[21:00:17.331] [Control] 普通弹框返回：1
[21:00:17.331] [Profile] 写入档案列表
[21:00:17.331] [Profile] 开始写入档案，共 0 个
[21:00:17.332] [Profile] 刷新档案列表
[21:00:17.332] [Profile] 开始获取档案列表
[21:00:17.332] [Profile] 获取档案完成，获取到 0 个档案
[21:00:17.332] [Profile] 档案列表刷新完成
[21:00:17.375] [UI] 弹出提示：档案删除成功！
[21:00:18.298] [Control] 按下图标按钮：BtnNew
[21:00:18.313] [Control] 选择弹窗：新建档案 - 选择验证类型
[21:00:19.186] [Control] 按下单选框：离线验证
[21:00:20.314] [Control] 按下单选框：第三方验证
[21:00:20.810] [Control] 按下单选框：正版验证
[21:00:21.602] [Control] 按下按钮：继续
[21:00:21.602] [Control] 选择弹框返回：1
[21:00:21.605] [Profile] 刷新档案列表
[21:00:21.605] [Profile] 开始获取档案列表
[21:00:21.606] [Profile] 获取档案完成，获取到 0 个档案
[21:00:21.606] [Profile] 档案列表刷新完成
[21:00:21.878] [Report] FPS 0, 动画 9, 下载中 0（0 B/s）
[21:00:24.370] [Control] 按下按钮：开始正版验证
[21:00:24.375] 正版登录尝试失败：索引超出范围。必须为非负值并小于集合大小。
参数名: index
   在 System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   在 PCL.ModProfile.GetLoginData()
   在 PCL.PageLoginMs._Lambda$__2-0()
   错误类型：System.ArgumentOutOfRangeException
[21:00:24.430] [Control] 普通弹窗：出现错误
正版登录尝试失败：索引超出范围。必须为非负值并小于集合大小。
参数名: index
   在 PCL.ModProfile.GetLoginData()
   在 PCL.PageLoginMs._Lambda$__2-0()
   错误类型：System.ArgumentOutOfRangeException
[21:01:12.948] [Report] FPS 0, 动画 8, 下载中 0（0 B/s）
[21:01:33.147] [Control] 按下按钮：确定
[21:01:34.521] [Control] 按下按钮：登录
[21:01:34.522] 正版登录尝试失败：索引超出范围。必须为非负值并小于集合大小。
参数名: index
   在 System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   在 PCL.ModProfile.GetLoginData()
   在 PCL.PageLoginMs._Lambda$__2-0()
   错误类型：System.ArgumentOutOfRangeException
[21:01:34.559] [Control] 普通弹窗：出现错误
正版登录尝试失败：索引超出范围。必须为非负值并小于集合大小。
参数名: index
   在 PCL.ModProfile.GetLoginData()
   在 PCL.PageLoginMs._Lambda$__2-0()
   错误类型：System.ArgumentOutOfRangeException
[21:01:35.266] [Control] 按下按钮：确定
[21:01:36.794] [Control] 按下文本按钮：«  返回
[21:01:36.795] [Profile] 刷新档案列表
[21:01:36.795] [Profile] 开始获取档案列表
[21:01:36.796] [Profile] 获取档案完成，获取到 0 个档案
[21:01:36.796] [Profile] 档案列表刷新完成
[21:01:37.690] [Control] 按下图标按钮：BtnNew
[21:01:37.696] [Control] 选择弹窗：新建档案 - 选择验证类型
[21:01:39.122] [Control] 按下单选框：第三方验证
[21:01:39.369] [Report] FPS 0, 动画 8, 下载中 0（0 B/s）
[21:01:39.537] [Control] 按下按钮：继续
[21:01:39.539] [Control] 选择弹框返回：2
[21:01:39.541] [Profile] 刷新档案列表
[21:01:39.541] [Profile] 开始获取档案列表
[21:01:39.541] [Profile] 获取档案完成，获取到 0 个档案
[21:01:39.541] [Profile] 档案列表刷新完成
[21:01:41.244] [Control] 按下按钮：登录
[21:01:41.258] [UI] 弹出提示：验证服务器、用户名与密码均不能为空！
[21:01:45.921] [Control] 按下文本按钮：
[21:01:45.922] [System] 正在打开网页：https://littleskin.cn/auth/register
[21:01:57.633] [Control] 按下文本按钮：
[21:01:57.633] [System] 正在打开网页：https://littleskin.cn/auth/register
[21:02:02.049] [Control] 按下按钮：返回
[21:02:02.050] [Profile] 刷新档案列表
[21:02:02.050] [Profile] 开始获取档案列表
[21:02:02.050] [Profile] 获取档案完成，获取到 0 个档案
[21:02:02.051] [Profile] 档案列表刷新完成
[21:02:02.953] [Control] 按下图标按钮：BtnNew
[21:02:02.988] [Control] 选择弹窗：新建档案 - 选择验证类型
[21:02:04.257] [Control] 按下按钮：取消
[21:02:04.259] [Control] 选择弹框返回：null
[21:02:04.261] [Profile] 刷新档案列表
[21:02:04.261] [Profile] 开始获取档案列表
[21:02:04.261] [Profile] 获取档案完成，获取到 0 个档案
[21:02:04.261] [Profile] 档案列表刷新完成
[21:02:04.873] [Control] 按下按钮：版本设置
[21:02:04.941] [MyPageLeft] 获取到 AnimatedControl 的值为 null
[21:02:04.946] [Control] 切换主要页面：VersionSetup, 0
[21:02:07.026] [Control] 按下单选列表项：设置
[21:02:08.060] [Control] 展开卡片：高级启动选项
[21:02:24.875] [Control] 按下图标按钮：BtnTitleInner
[21:02:24.876] [Control] 切换主要页面：Launch, -1
[21:02:25.023] [Page] 主页预设：Minecraft 新闻
[21:02:25.023] [Page] 主页自定义数据来源：联网缓存文件
[21:02:25.045] [MyPageLeft] 获取到 AnimatedControl 的值为 null
[21:02:26.250] [Control] 按下图标按钮：BtnNew
[21:02:26.296] [Control] 选择弹窗：新建档案 - 选择验证类型
[21:02:27.081] [Control] 按下单选框：第三方验证
[21:02:27.610] [Control] 按下按钮：继续
[21:02:27.611] [Control] 选择弹框返回：2
[21:02:27.611] [Profile] 刷新档案列表
[21:02:27.611] [Profile] 开始获取档案列表
[21:02:27.611] [Profile] 获取档案完成，获取到 0 个档案
[21:02:27.611] [Profile] 档案列表刷新完成
[21:02:37.907] [Report] FPS 0, 动画 6, 下载中 0（0 B/s）
[21:02:54.957] [Profile] 刷新档案列表
[21:02:54.957] [Profile] 开始获取档案列表
[21:02:54.958] [Profile] 获取档案完成，获取到 0 个档案
[21:02:54.958] [Profile] 档案列表刷新完成
[21:02:56.993] [Control] 按下图标按钮：BtnNew
[21:02:57.036] [Control] 选择弹窗：新建档案 - 选择验证类型
[21:02:58.210] [Control] 按下单选框：第三方验证
[21:02:58.666] [Control] 按下按钮：继续
[21:02:58.666] [Control] 选择弹框返回：2
[21:02:58.666] [Profile] 刷新档案列表
[21:02:58.666] [Profile] 开始获取档案列表
[21:02:58.666] [Profile] 获取档案完成，获取到 0 个档案
[21:02:58.666] [Profile] 档案列表刷新完成
[21:03:25.153] [Control] 按下按钮：登录
[21:03:25.154] [Loader] 加载器 Loader Login Auth 状态改变：Loading
[21:03:25.155] [Launch] 验证方式：Authlib-Injector（1012737146@qq.com）
[21:03:25.155] [Launch] 登录开始（Login, Auth）
[21:03:25.155] [Net] 发起网络请求（POST，https://skin.yudream.online/api/yggdrasil/authserver/authenticate），最大超时 15000
[21:03:25.155] [Net] 当前代理状态：禁用
[21:03:25.652] [Net] 获取网络结果：https://skin.yudream.online/api/yggdrasil，超时 10000ms
[21:03:25.946] [Profile] 写入档案列表
[21:03:25.957] 写入档案列表失败：String 引用没有设置为 String 的实例。
参数名: s
   在 System.Text.Encoding.GetBytes(String s)
   在 PCL.ModSecret.SecretEncrypt(String SourceString)
   在 PCL.ModProfile.WriteProfileJson(JArray ListJson)
   错误类型：System.ArgumentNullException
[21:03:26.018] [Control] 普通弹窗：出现错误
写入档案列表失败：String 引用没有设置为 String 的实例。
参数名: s
   在 PCL.ModSecret.SecretEncrypt(String SourceString)
   在 PCL.ModProfile.WriteProfileJson(JArray ListJson)
   错误类型：System.ArgumentNullException

是否反馈此问题？如果不反馈，这个问题可能永远无法得到解决！
[21:03:32.524] [Control] 按下按钮：取消
[21:03:32.524] [Control] 普通弹框返回：2
[21:03:32.524] [Launch] 登录成功（Login, Auth）
[21:03:32.524] 保存启动记录失败：未找到设置项：LoginAuthEmail
   在 PCL.ModSetup.Get(String Key, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:32.524] 加载线程 Loader Login Auth (26) 出错，已完成 95%：给定关键字不在字典中。
   在 System.ThrowHelper.ThrowKeyNotFoundException()
   在 System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   在 PCL.ModSetup.Set(String Key, Object Value, Boolean ForceReload, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:32.524] [Loader] 加载器 Loader Login Auth 状态改变：Failed
[21:03:32.575] 第三方登录尝试失败：给定关键字不在字典中。
   在 PCL.PageLoginAuth._Closure$__5-0._Lambda$__0()
→ 给定关键字不在字典中。
   在 System.ThrowHelper.ThrowKeyNotFoundException()
   在 System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   在 PCL.ModSetup.Set(String Key, Object Value, Boolean ForceReload, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:32.707] [Control] 普通弹窗：出现错误
第三方登录尝试失败：给定关键字不在字典中。
   在 PCL.PageLoginAuth._Closure$__5-0._Lambda$__0()
→ 给定关键字不在字典中。
   在 PCL.ModSetup.Set(String Key, Object Value, Boolean ForceReload, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:34.306] [Control] 按下按钮：确定
[21:03:35.146] [Control] 按下按钮：登录
[21:03:35.147] [Loader] 加载器 Loader Login Auth 状态改变：Loading
[21:03:35.147] [Launch] 验证方式：Authlib-Injector（1012737146@qq.com）
[21:03:35.147] [Launch] 登录开始（Login, Auth）
[21:03:35.147] [Net] 发起网络请求（POST，https://skin.yudream.online/api/yggdrasil/authserver/authenticate），最大超时 15000
[21:03:35.147] [Net] 当前代理状态：禁用
[21:03:35.880] [Net] 获取网络结果：https://skin.yudream.online/api/yggdrasil，超时 10000ms
[21:03:36.111] [Profile] 写入档案列表
[21:03:36.120] 写入档案列表失败：String 引用没有设置为 String 的实例。
参数名: s
   在 System.Text.Encoding.GetBytes(String s)
   在 PCL.ModSecret.SecretEncrypt(String SourceString)
   在 PCL.ModProfile.WriteProfileJson(JArray ListJson)
   错误类型：System.ArgumentNullException
[21:03:36.168] [Control] 普通弹窗：出现错误
写入档案列表失败：String 引用没有设置为 String 的实例。
参数名: s
   在 PCL.ModSecret.SecretEncrypt(String SourceString)
   在 PCL.ModProfile.WriteProfileJson(JArray ListJson)
   错误类型：System.ArgumentNullException

是否反馈此问题？如果不反馈，这个问题可能永远无法得到解决！
[21:03:36.909] [Control] 按下按钮：取消
[21:03:36.909] [Control] 普通弹框返回：2
[21:03:36.909] [Launch] 登录成功（Login, Auth）
[21:03:36.909] 保存启动记录失败：未找到设置项：LoginAuthEmail
   在 PCL.ModSetup.Get(String Key, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:36.909] 加载线程 Loader Login Auth (28) 出错，已完成 95%：给定关键字不在字典中。
   在 System.ThrowHelper.ThrowKeyNotFoundException()
   在 System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   在 PCL.ModSetup.Set(String Key, Object Value, Boolean ForceReload, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:36.909] [Loader] 加载器 Loader Login Auth 状态改变：Failed
[21:03:36.935] 第三方登录尝试失败：给定关键字不在字典中。
   在 PCL.PageLoginAuth._Closure$__5-0._Lambda$__0()
→ 给定关键字不在字典中。
   在 System.ThrowHelper.ThrowKeyNotFoundException()
   在 System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   在 PCL.ModSetup.Set(String Key, Object Value, Boolean ForceReload, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:37.097] [Control] 普通弹窗：出现错误
第三方登录尝试失败：给定关键字不在字典中。
   在 PCL.PageLoginAuth._Closure$__5-0._Lambda$__0()
→ 给定关键字不在字典中。
   在 PCL.ModSetup.Set(String Key, Object Value, Boolean ForceReload, McVersion Version)
   在 PCL.ModLaunch.McLoginServerStart(LoaderTask`2 Data)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
   错误类型：System.Collections.Generic.KeyNotFoundException
[21:03:37.738] [Control] 按下按钮：确定
[21:04:27.658] [Control] 按下按钮：返回
[21:04:27.661] [Profile] 刷新档案界面
[21:04:27.661] [Profile] 使用 Authlib 皮肤加载器
[21:04:27.662] [Loader] 加载器 Loader Skin Auth 状态改变：Loading
[21:04:27.664] [Net] 获取网络结果：https://skin.yudream.online/api/yggdrasil/sessionserver/session/minecraft/profile/27df4c20edce32c18c7f678020360852，超时 10000ms
[21:04:28.113] [Skin] UUID 27df4c20edce32c18c7f678020360852 对应的皮肤文件为 https://skin.yudream.online/textures/564c51dcf3e48f7b6709d8c2efefc64ecc9a2e23c1850e66d515adfb6d0284f6
[21:04:28.114] [Net] 直接下载文件：https://skin.yudream.online/textures/564c51dcf3e48f7b6709d8c2efefc64ecc9a2e23c1850e66d515adfb6d0284f6
[21:04:28.394] [Minecraft] 皮肤下载成功：C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\4872446093047215153.png
[21:04:28.394] [Loader] 加载器 Loader Skin Auth 状态改变：Finished
[21:04:28.411] [Animation] 两个动画帧间隔 734 ms
[21:04:28.415] 载入头像失败（C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\4872446093047215153.png,Loader Skin Auth）：未能找到路径“C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\Head\Auth_xin_yi_hl_27df4c20edce32c18c7f678020360852.png”的一部分。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   在 PCL.MySkin.Load()
   错误类型：System.IO.DirectoryNotFoundException
[21:04:31.449] [Control] 按下图标按钮：BtnSelect
[21:04:31.451] [Profile] 刷新档案列表
[21:04:31.451] [Profile] 开始获取档案列表
[21:04:31.451] [Profile] 获取档案完成，获取到 0 个档案
[21:04:31.451] [Profile] 档案列表刷新完成
[21:04:33.850] [Control] 按下图标按钮：BtnNew
[21:04:33.883] [Control] 选择弹窗：新建档案 - 选择验证类型
[21:04:34.803] [Control] 按下单选框：正版验证
[21:04:35.138] [Control] 按下单选框：离线验证
[21:04:35.691] [Control] 按下按钮：继续
[21:04:35.691] [Control] 选择弹框返回：0
[21:04:35.905] [Control] 输入弹窗：新建档案 - 输入档案名称
[21:04:41.538] [Control] 按下按钮：继续
[21:04:41.538] [Control] 输入弹框返回：xin_yi_hl
[21:04:41.735] [Control] 选择弹窗：新建档案 - 选择 UUID 类型
[21:04:42.914] [Control] 按下单选框：行业规范 UUID（推荐）
[21:04:43.498] [Control] 按下按钮：继续
[21:04:43.498] [Control] 选择弹框返回：0
[21:04:43.498] [Profile] 写入档案列表
[21:04:43.498] [Profile] 开始写入档案，共 1 个
[21:04:43.499] [Profile] 刷新档案列表
[21:04:43.499] [Profile] 开始获取档案列表
[21:04:43.499] [Profile] 获取档案完成，获取到 1 个档案
[21:04:43.500] [Profile] 档案列表刷新完成
[21:04:43.525] [UI] 弹出提示：档案新建成功！
[21:04:44.932] [Profile] 选定档案: xin_yi_hl, 以 Legacy 方式验证
[21:04:44.932] [Profile] 刷新档案界面
[21:04:44.932] [Profile] 使用离线皮肤加载器
[21:04:44.932] [Loader] 加载器 Loader Skin Legacy 状态改变：Loading
[21:04:44.933] [Loader] 加载器 Loader Skin Legacy 状态改变：Finished
[21:04:44.945] [Control] 按下单击列表项：xin_yi_hl
[21:04:44.947] 载入头像失败（pack://application:,,,/Plain Craft Launcher 2;component/Images/Skins/Steve.png,Loader Skin Legacy）：未能找到路径“C:\Users\xinyihl\AppData\Local\Temp\PCL\Cache\Skin\Head\Legacy_xin_yi_hl_27df4c20edce32c18c7f678020360852.png”的一部分。
   在 System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   在 System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   在 PCL.MySkin.Load()
   错误类型：System.IO.DirectoryNotFoundException

``` 

</details>

Pigeon0v0 commented:
额 如果可以的话最好补充一下复现步骤方便调试 = =

xinyihl commented:
> 额 如果可以的话最好补充一下复现步骤方便调试 = =

https://airportal.cn/450190/fGCIFmbfeI

Pigeon0v0 committed:
feat: 添加第三方验证档案时显示验证服务器名称
fix: 新建档案文件没有 Close 流
fix: 意外允许在无档案时导出档案
fix: 新建档案时意外尝试获取当前档案信息，但当前档案其实是炒饭
fix: 加解密字符串的传入字符串可能是炒饭
imp: 优化头像存储，避免潜在的多验证服务器冲突
dev: clean up

Pigeon0v0 committed:
feat: 档案列表界面提示
chore: 现在不允许在选择 UUID 步骤取消操作
fix: 没有游戏时也禁用启动按钮

Pigeon0v0 committed:
dev: reduce warning

Pigeon0v0 committed:
fix: 启动画面不会切换

Pigeon0v0 committed:
feat: 加入了第三方验证的快速注册按钮
refactor: 头像缓存更换文件名
dev: 移除统一通行证相关代码

Pigeon0v0 committed:
fix: 没检测有没有文件夹

wyc-26 committed:
Merge branch 'dev' into profile

Pigeon0v0 committed:
feat: 自动迁移旧版档案，需要等 2.11.0 发版时修改 FrmMain
imp: 使用 ProfileLog 方法记录档案相关日志
fix: 正版验证日志步骤编号写错了
chore: clean up

xinyihl commented:
点击导入依然会爆
[Log-CE1.log](https://github.com/user-attachments/files/20000376/Log-CE1.log)
正版登录还是不行
[Log-CE1.log](https://github.com/user-attachments/files/20000391/Log-CE1.log)
第三方和头像没问题了

Pigeon0v0 committed:
fix: 导入第三方验证档案可能失败
fix: 可能意外设置 lastUsed 值为 -1

Pigeon0v0 commented:
> 点击导入依然会爆

Fixed in 5d9fd2b

> 正版登录还是不行

这可能是因为这个 PR 还没被推送到主仓库，推送了就好了

Pigeon0v0 committed:
Merge branch 'dev' into profile

xinyihl commented:
> > 点击导入依然会爆
> 
> Fixed in [5d9fd2b](https://github.com/PCL-Community/PCL2-CE/commit/5d9fd2bea9683ae6fcfced40d32cc2ca5b51c823)
> 
> > 正版登录还是不行
> 
> 这可能是因为这个 PR 还没被推送到主仓库，推送了就好了
> 
> @xinyihl 再 Review 一下？

导入ok了
正版登录还是不行

Pigeon0v0 commented:
> 正版登录还是不行
> 
> 我看报错是请求体没有'client_id'

正常的 大概吧

Pigeon0v0 committed:
Merge branch 'dev' into profile
